File: week-5/boridevna/완주하지못한선수.py
def solution(participant, completion):
    if len(completion) == 0:
        return participant[0]

    check_dict = dict()

    for p in participant:
        if p not in check_dict:
            check_dict[p] = 1
        else:
            check_dict[p] += 1

    for c in completion:
        if check_dict[c] > 1:
            check_dict[c] -= 1
        else:
            del check_dict[c]

    return check_dict.popitem()[0]

File: week-5/newbie/완주하지못한선수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    sort(participant.begin(), participant.end());
    sort(completion.begin(), completion.end());
    
    for (int i = 0; i < completion.size(); i++) {
        if (participant[i] != completion[i])
            return participant[i];
    }
    return participant[participant.size() - 1];
}
